system/jlib/jtrace.cpp

```diff
@@ -69,3 +100,617 @@ const char* LogTrace::queryLocalId() const
+     */
+    /*bool injectSpanContext(CHPCCHttpTextMapCarrier * carrier) override
+    {
+        if (!carrier)
+            return false;
+
+        auto propagator = opentelemetry::context::propagation::GlobalTextMapPropagator::GetGlobalPropagator();
+
+        //get current context...
+        opentelemetry::v1::context::Context currentCtx = opentelemetry::context::RuntimeContext::GetCurrent();
+        //and have the propagator inject the ctx into carrier
+        propagator->Inject(*carrier, currentCtx);
+
+        if (!isEmptyString(hpccGlobalId.get()))
+            carrier->Set(HPCCSemanticConventions::kGLOBALIDHTTPHeader, hpccGlobalId.get());
+
+        if (!isEmptyString(hpccCallerId.get()))
+            carrier->Set(HPCCSemanticConventions::kCallerIdHTTPHeader, hpccCallerId.get());
+
+        return true;
+    }*/
+
+    bool injectSpanContext(IProperties * contextProps) override
+    {
+        if (!contextProps || isEmptyString(traceID.get()) || isEmptyString(spanID.get()) || isEmptyString(traceFlags.get()))
+            return false;
+
+        StringBuffer contextHTTPHeader;
+        //The traceparent header uses the version-trace_id-parent_id-trace_flags format where:
+        //version is always 00. trace_id is a hex-encoded trace id. span_id is a hex-encoded span id. trace_flags is a hex-encoded 8-bit field that contains tracing flags such as sampling, trace level, etc.
+        //Example: "traceparent", "00-beca49ca8f3138a2842e5cf21402bfff-4b960b3e4647da3f-01"
+        contextHTTPHeader.append("00-").append(traceID.get()).append("-").append(spanID.get()).append(traceFlags.get());
+        contextProps->setProp(opentelemetry::trace::propagation::kTraceParent.data(), contextHTTPHeader.str());
+
+        StringBuffer traceStateHTTPHeader;
+        traceStateHTTPHeader.append("hpcc=").append(spanID.get());
+
+        contextProps->setProp(opentelemetry::trace::propagation::kTraceState.data(), contextHTTPHeader.str());
+
+        if (!isEmptyString(hpccGlobalId.get()))
+            contextProps->setProp(HPCCSemanticConventions::kGLOBALIDHTTPHeader, hpccGlobalId.get());
+
+        if (!isEmptyString(hpccCallerId.get()))
+            contextProps->setProp(HPCCSemanticConventions::kCallerIdHTTPHeader, hpccCallerId.get());
+
+        return true;
+    }
+
+    opentelemetry::v1::trace::SpanContext querySpanContext() const
+    {
+        if (span != nullptr)
+            return span->GetContext();
+
+        return opentelemetry::trace::SpanContext::GetInvalid();
+    }
+
+    const char * queryTraceName() const
+    {
+        return tracerName.get();
+    }
+
+protected:
+    CSpan(const char * spanName, SpanType type, CSpan * parent)
+    {
+        name.set(spanName);
+        parentSpan = parent;
+        if (parentSpan != nullptr)
+            tracerName.set(parent->queryTraceName());
+        this->type = type;
+
+         init();
+    }
+
+    CSpan(const char * spanName, SpanType type, const char * tracerName_)
```

Comment:
consistency: 3 different ways of differentiating the parameter from the member i) diff name ii) use this for the member iii) append an underscore.
I don't really mind any of them (although everywhere else uses _ as a prefix rather than a suffix), but much better to be consistent.

Reply:
one of the many reasons I prefer to prefix all members with m_